apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -103,4 +106,40 @@ export const resourceRouter = router({
         ])
         .execute()
     }),
+
+  getWithFullPermalink: protectedProcedure
+    .input(getFullPermalinkSchema)
+    .query(async ({ input: { resourceId } }) => {
+      return db
+        .withRecursive("resourcePath", (eb) =>
+          eb
+            .selectFrom("Resource as r")
+            .select([
+              "r.id",
+              "r.title",
+              "r.permalink",
+              "r.parentId",
+              "r.permalink as fullPermalink",
+            ])
+            .where("r.parentId", "is", null)
+            .unionAll(
+              eb
+                .selectFrom("Resource as s")
+                .innerJoin("resourcePath as rp", "s.parentId", "rp.id")
+                .select([
+                  "s.id",
+                  "s.title",
+                  "s.permalink",
+                  "s.parentId",
+                  sql<string>`CONCAT(rp."fullPermalink", '/', s.permalink)`.as(
```

Comment:
should we use functions instead of raw sql?

```ts
   fn<string>('concat', [
      'rp.fullPermalink',
      val('/'),
      's.permalink'
    ]).as('fullPermalink'),
```

See [kysely docs](https://kysely.dev/docs/examples/select/function-calls)

Reply:
For some reason couldn't get it to work, so kept it as raw sql for now since this is relatively straightforward and not a significant difference from the functions way.